plugins/obs-filters/drop-shadow-filter.c

```diff
@@ -0,0 +1,400 @@
+	}
+
+	bfree(data);
+}
+
+static void *drop_shadow_create(obs_data_t *settings, obs_source_t *context)
+{
+	struct drop_shadow_filter_data *filter =
+			bzalloc(sizeof(struct drop_shadow_filter_data));
+	char *effect_path = obs_module_file("drop_shadow_filter.effect");
+
+	filter->context = context;
+
+	obs_enter_graphics();
+
+	filter->shadow1_target = gs_texrender_create(GS_RGBA, GS_ZS_NONE);
+	filter->shadow2_target = gs_texrender_create(GS_RGBA, GS_ZS_NONE);
+
+	char *error_string;
+	filter->effect = gs_effect_create_from_file(effect_path, &error_string);
+	if (filter->effect) {
+		filter->offset_param = gs_effect_get_param_by_name(
+				filter->effect, "offset");
+		filter->shadow_color_param = gs_effect_get_param_by_name(
+				filter->effect, "shadow_color");
+		filter->image_param = gs_effect_get_param_by_name(
+				filter->effect, "image");
+		filter->image_uv_offset_param = gs_effect_get_param_by_name(
+				filter->effect, "image_uv_offset");
+		filter->image_uv_scale_param = gs_effect_get_param_by_name(
+				filter->effect, "image_uv_scale");
+		filter->image_uv_pixel_interval_param =
+				gs_effect_get_param_by_name(
+				filter->effect, "image_uv_pixel_interval");
+		filter->shadow_uv_pixel_interval_param =
+				gs_effect_get_param_by_name(
+				filter->effect, "shadow_uv_pixel_interval");
+		filter->shadow_param = gs_effect_get_param_by_name(
+				filter->effect, "shadow");
+		filter->kawase_distance_param = gs_effect_get_param_by_name(
+				filter->effect, "kawase_distance");
+	} else {
+		fprintf(stderr, "Failed to compile '%s':\n%s\n",
+				effect_path, error_string);
+	}
+
+	obs_leave_graphics();
+
+	bfree(effect_path);
+
+	if (!filter->effect) {
+		drop_shadow_destroy(filter);
+		return NULL;
+	}
+
+	drop_shadow_update(filter, settings);
+	return filter;
+}
+
+static void set_parameters(
+		struct drop_shadow_filter_data *filter,
+		gs_texrender_t *shadow_source,
+		float kawase_distance)
+{
+	gs_texture_t *shadow = gs_texrender_get_texture(
+			shadow_source);
+
+	gs_effect_set_vec2(filter->offset_param, &filter->offset);
+	gs_effect_set_vec4(filter->shadow_color_param, &filter->shadow_color);
+	gs_effect_set_vec2(filter->image_uv_offset_param,
+			&filter->image_uv_offset);
+	gs_effect_set_vec2(filter->image_uv_scale_param,
+			&filter->image_uv_scale);
+	gs_effect_set_vec2(filter->image_uv_pixel_interval_param,
+			&filter->image_uv_pixel_interval);
+	gs_effect_set_vec2(filter->shadow_uv_pixel_interval_param,
+			&filter->shadow_uv_pixel_interval);
+	gs_effect_set_float(filter->kawase_distance_param,
+			kawase_distance);
+	gs_effect_set_texture(filter->shadow_param, shadow);
+}
+
+static void set_image_parameter(struct drop_shadow_filter_data *filter)
+{
+	gs_texture_t *image = obs_filter_get_texture(filter->context);
+	gs_effect_set_texture(filter->image_param, image);
+}
+
+typedef struct {
+	int passes;
+	float distances[7];
+} kawase_entry_t;
+
+/* In the kawase blur algorithm four texture samples are taken in
```

Comment:
In comments, please capitalize "Kawase" as it is a proper name.  Variable names can follow whatever naming/capitalization convention being used (snake_case, camelCase, etc.).  Be sure to check this throughout.

Reply:
Done.